packages/trip-form/src/util.js

```diff
@@ -69,36 +116,88 @@ export function getTransitSubmodeOptions(icons, modes, selectedModes) {
   });
 }
 
-function getPrimaryModeOption(icons, selectedModes) {
+/**
+ * Returns big primary "Take Transit" choice.
+ * @param {*} ModeIcon The icon component for rendering.
+ * @param {*} selectedModes An array of string that lists the modes selected for a trip query.
+ */
+function getPrimaryModeOption(ModeIcon, selectedModes) {
   return {
     id: "TRANSIT",
     selected: selectedModes.some(isTransit) && selectedModes.includes("WALK"),
     showTitle: false,
     text: (
       <span>
-        <ModeIcon icons={icons} mode="TRANSIT" />
+        <ModeIcon mode="TRANSIT" />
```

Comment:
In general, I'm not too sure about the idea of removing the ability to supply custom icons for the `ModeIcon` component. It seems like that was an important part of handling custom stuff in trimet-mod-otp (e.g., https://github.com/ibi-group/trimet-mod-otp/blob/septa/lib/config.js#L24-L36), but it's been a while since I thought about how all of the pieces fit together, so I may be off there.

Reply:
It looks @evansiroky has ported that logic to `TriMetLegIcon` in the early days at https://github.com/opentripplanner/otp-ui/blob/master/packages/icons/src/trimet-leg-icon.js. The idea is if an agency has custom logic they would supply their own ModeIcon and/or or LegIcon components. Is this used anywhere else beside legs?

Reply:
I'm not sure... is it used in the SettingsSelector, @binh-dam-ibigroup?

Reply:
I these methods, I replaced parameter `icons` with `ModeIcon` and removed the component of that name. The call highlighted is using the parameter.